### PR TITLE
Try wrapping entire coming soon page under cover block to fix layout issue

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Another attempt to partially fix layout issues from https://github.com/woocommerce/team-ghidorah/issues/311 by wrapping everything under the cover block and flexbox positioning instead of using absolute positioning.

_Note that this does not yet fix the styling issues (such as changing background color, default margin/padding mismatches, etc.)_

### How to test the changes in this Pull Request:

1. Enable `launch-your-store` feature flag.
2. Go to `WooCommerce > Settings > Site Visibility` and enable `Coming soon` checkbox and save.
3. Go to `Appearance > Editor > Templates > Page: Coming soon` and edit it.
4. Try adding blocks around and see if the frontend layout matches.

https://github.com/woocommerce/woocommerce/assets/3747241/f9522cca-24d0-4829-b0d0-217b0a7e7302

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/coming-soon/edit.tsx [hunk 1]
```diff
@@ -118,16 +115,23 @@ export default function Edit( { attributes, setAttributes } ) {
...
                 .woocommerce-coming-soon-banner-container {
-                    padding-inline: min(6.5rem, 8vw);
+                    padding-inline: min(5.5rem, 8vw);
                     margin: 0;
+                    height: 100%;
+                    display: flex;
+                    flex-direction: column;
+                    justify-content: space-between;
+                }
+                .woocommerce-coming-soon-banner-container > .wp-block-group__inner-container {
```
Comment (ilyasfoo) [inline plugins/woocommerce-blocks/assets/js/blocks/coming-soon/edit.tsx]: This is required for classic themes, `wp-block-group__inner-container` appears out of nowhere 🤷

File: plugins/woocommerce/patterns/coming-soon-entire-site.php [hunk 2]
```diff
@@ -156,15 +173,12 @@
...
-	transform: translateY(-50%);
-	margin-block-start: 0;
+.coming-soon-is-vertically-aligned-center {
 	width: 100%;
+	align-items: stretch;
 }
-.woocommerce-coming-soon-header, .coming-soon-cover .wp-block-cover__background {
-	background-color: #bea0f2 !important;
+.woocommerce-coming-soon-header {
+	height: 40px;
```
Comment (ilyasfoo) [inline plugins/woocommerce/patterns/coming-soon-entire-site.php:184]: Small hack: Fixed header height in order to ensure small modifications will keep the center part relatively stable

File: plugins/woocommerce-blocks/assets/js/blocks/coming-soon/index.tsx [hunk 3]
```diff
@@ -135,15 +139,12 @@ registerBlockType( metadata, {
...
-                    position: absolute;
-                    top: 50%;
-                    transform: translateY(-50%);
-                    margin-block-start: 0;
+                .coming-soon-is-vertically-aligned-center {
                     width: 100%;
+                    align-items: stretch;
                 }
-                .woocommerce-coming-soon-header, .coming-soon-cover .wp-block-cover__background {
-                    background-color: ${ color } !important;
```
Comment (psealock) [inline plugins/woocommerce-blocks/assets/js/blocks/coming-soon/index.tsx]: With the Coming soon block selected, the background color is not longer respecting the selection

image

**Review by psealock — COMMENTED** [review]
This is working nicely @ilyasfoo.

I have [a bit of a refactor currently under review](https://github.com/woocommerce/woocommerce/pull/46854) as well which will cause some conflicts. Do you want to give it a shot here or after merge?

**Comment by ilyasfoo** [comment]
> I have https://github.com/woocommerce/woocommerce/pull/46854 as well which will cause some conflicts. Do you want to give it a shot here or after merge?

@psealock  Thanks for the heads up! I think it'll be better to handle after the PR is merged. Would you have that merged soon? I think it looks good to merge with 1 review

File: plugins/woocommerce-blocks/assets/js/blocks/coming-soon/index.tsx (hunk 3 above)
Comment (ilyasfoo) [reply plugins/woocommerce-blocks/assets/js/blocks/coming-soon/index.tsx]: Good catch!

**Comment by psealock** [comment]
I merged https://github.com/woocommerce/woocommerce/pull/46854 @ilyasfoo, let me know if any issues arise

**Comment by ilyasfoo** [comment]
> I merged https://github.com/woocommerce/woocommerce/pull/46854 @ilyasfoo, let me know if any issues arise

@psealock Thanks! I've resolved conflicts without issues

~~Edit: Actually, I noticed a bug with rendering store-only coming soon. Will investigate next week~~

Edit 2: Hmm, false alarm. Tested again in a fresh JN and no issues found. This PR is ready for review!

File: plugins/woocommerce-blocks/assets/js/blocks/coming-soon/index.tsx (hunk 3 above)
Comment (ilyasfoo) [reply plugins/woocommerce-blocks/assets/js/blocks/coming-soon/index.tsx]: Fixed in https://github.com/woocommerce/woocommerce/pull/46914/commits/c471e520d322b7073fa4ae854d7113be0b7c92a5

**Review by psealock — APPROVED** [review]
This is looking great @ilyasfoo, testing well and code changes look good 🚢

**Review by chihsuan — APPROVED** [review]
Thanks @ilyasfoo. Nice fix.  🚢

**Pull request merged**